src/main/java/backend/resource/TurboLabel.java

```diff
@@ -197,18 +228,24 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         TurboLabel that = (TurboLabel) o;
-        return actualName.equals(that.actualName) && colour.equals(that.colour);
+        return fullName.equals(that.fullName) && colour.equals(that.colour);
     }
 
     @Override
     public int hashCode() {
-        int result = actualName.hashCode();
+        int result = fullName.hashCode();
         result = 31 * result + colour.hashCode();
         return result;
     }
 
     @Override
     public int compareTo(TurboLabel o) {
-        return actualName.compareTo(o.getActualName());
+        return fullName.compareTo(o.getFullName());
+    }
+
+    private Grouping determineGrouping(String delimiter) {
+        if (delimiter.isEmpty()) return Grouping.NONE;
+
+        return delimiter.equals(EXCLUSIVE_DELIMITER) ? Grouping.EXCLUSIVE : Grouping.NON_EXCLUSIVE;
```

Comment:
Assumes only three possibilities, but does not check for it. Use a `switch` with a `default: assert false`?